connectionRequester/connectionRequester.go

```diff
@@ -0,0 +1,191 @@
+package connectionRequester
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/godbus/dbus"
+)
+
+const (
+	dbusPath   = "/org/cacophony/modemd"
+	dbusDest   = "org.cacophony.modemd"
+	methodBase = "org.cacophony.modemd"
+
+	requestinterval = 20
+	wifiInterface   = "wlan0"
+	pingTimeout     = 5
+)
+
+var hosts = []string{
+	"8.8.8.8",
+	"8.8.4.4",
+}
+
+func sendOnRequest() error {
+	obj, err := getDbusObj()
+	if err != nil {
+		return err
+	}
+	return obj.Call(methodBase+".StayOn", 0).Store()
+}
+
+type ConnectionRequester struct {
+	stateChange  chan bool
+	sendRequests bool
+}
+
+// NewConnectionRequester will return a ConnectionRequester.
+// No connection will be requested until Start is called
+func NewConnectionRequester() *ConnectionRequester {
+	cr := &ConnectionRequester{
+		stateChange:  make(chan bool),
+		sendRequests: false,
+	}
+	go cr.requestConnections()
+	return cr
+}
+
+// WaitUntilUp will wait until a connection has been made returning an error
+// if no connection is made in the given duration
+func (cr *ConnectionRequester) WaitUntilUp(timeout time.Duration) error {
+	connectionTimeout := time.After(timeout)
+	for {
+		if CheckConnection() {
+			return nil
+		}
+		select {
+		case <-connectionTimeout:
+			return errors.New("connection failed")
+		case <-time.After(time.Second):
+		}
+	}
+}
+
+// WaitUntilUpLoop will wait until a connection has been made returning an error
+// if no connection is made.
+// timeout is the time given to make a connection each try.
+// retryAfter is the duration between attempts, it will get doubled after each try.
+// retryAfter is how many times it will try to make a connection. If -1 it will
```

Comment:
it's often a good idea to have a cap on retry interval so it doesn't get too crazy. 30 mins?